core/src/avm1/object.rs

```diff
@@ -128,16 +165,19 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
     ) -> Result<(), Error<'gc>>;
 
     /// Set a named property on this object, or its prototype.
+    #[allow(unused_mut)] //it's not unused
```

Comment:
I'm not sure why clippy thinks `value` does not need to be mutable, but there's another workaround that doesn't need `#[allow(unused_mut)]`:
```rs
let mut value = value;
```